core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestPushProjectionIntoTableScan.java

```diff
@@ -185,10 +188,18 @@ public void testPushProjection()
                                                 .build()))));
                     })
                     .matches(project(
-                            newNames.entrySet().stream()
+                            Stream.concat(newNames.entrySet().stream(), constants.entrySet().stream())
                                     .collect(toImmutableMap(
                                             e -> e.getKey().getName(),
-                                            e -> expression(symbolReference(e.getValue())))),
+                                            e -> {
+                                                if (e.getValue() instanceof String) {
+                                                    return expression(symbolReference((String) e.getValue()));
+                                                }
+                                                else if (e.getValue() instanceof Expression) {
```

Comment:
The "else" here is redundant since the "if" above always returns. So this can be
```java
if (e.getValue() instanceof Expression value) {
    return expression(value.getValue());
}

Reply:
done